python/langsmith/client.py

```diff
@@ -931,6 +933,24 @@ def request_with_retries(
                     filler = "*" * (max(0, len(api_key) - 7))
                     masked_api_key = f"{prefix}{filler}{suffix}"
 
+                    debug_crash_dump_file = ls_utils.get_env_var(
+                        "LANGSMITH_DEBUG_CRASH_DUMP"
+                    )
+                    if debug_crash_dump_file is not None and e.request is not None:
+                        try:
+                            request_data: (
+                                requests.Request | requests.PreparedRequest
+                            ) = copy.deepcopy(e.request)
+                            if "x-api-key" in request_data.headers:
+                                request_data.headers["x-api-key"] = masked_api_key
+                            with gzip.open(debug_crash_dump_file, "ab") as f:
```

Comment:
```suggestion
                            with gzip.open(debug_crash_dump_file, "ab", encoding="utf-8") as f:
```

Reply:
Ah sorry I lost this

Reply:
Actually quick question here @hinthornw 

Previously I was opening the files in byte mode "ab", and then directly writing in byte mode with .encode("utf-8") on my json.dumps.

With the above change, I need to open the file in text mode "at", and then just write strings back. Is this considered safer style?